rules/DeadCode/Rector/Assign/RemoveDoubleAssignRector.php

```diff
@@ -116,6 +103,11 @@ public function refactor(Node $node): ?Node
                 continue;
             }
 
+            // next stmts can have side effect as well
+            if ($nextAssign->expr instanceof MethodCall) {
```

Comment:
I think this needs more specific check, eg:

```suggestion
            if (($nextAssign->var instanceof PropertyFetch || $nextAssign->var instanceof StaticPropertyFetch)  && $nextAssign->expr instanceof MethodCall) {
```

like in line 111, I will create new PR for it

Reply:
https://github.com/rectorphp/rector-src/pull/6443